src/lib/post.ts

```diff
@@ -92,5 +94,27 @@ const getAllPosts = (fields: Field[] = []): Post[] => {
   return posts;
 };
 
-export { getPostBySlug, getAllPosts, POST_FIELDS };
+const randomPostSlug = (
+  excludedSlugs: string[] = []
+): { slug: string; excludedSlugs: string[] } => {
+  const slugs = getPostSlugs().filter((slug) => !excludedSlugs.includes(slug));
+
+  if (slugs.length === 0) {
+    return { slug: '', excludedSlugs };
+  }
+
+  const index = Math.floor(Math.random() * slugs.length);
+  const slug = slugs[index];
+  excludedSlugs.push(slug);
+
+  return { slug, excludedSlugs };
```

Comment:
@hoangmirs , could you explain a bit about the role of `excludedSlugs` here?

Reply:
I removed it out in https://github.com/nimblehq/til-web/pull/33/commits/38e21ea6c87731a4580902e22d23475ca1d2a5f7